src/main/java/net/minecraftforge/eventbus/EventBus.java

```diff
@@ -199,6 +199,30 @@ public <T extends Event> void addListener(final EventPriority priority, final bo
         addListener(priority, passCancelled(receiveCancelled), consumer);
     }
 
+    @Override
+    public <T extends Event> void addListener(EventPriority priority, Class<T> eventType, Consumer<T> consumer) {
+        checkNotGeneric(eventType);
+        addListener(priority, false, eventType, consumer);
+    }
+
+    @Override
+    public <T extends Event> void addListener(boolean receiveCancelled, Consumer<T> consumer) {
+        checkNotGeneric(consumer);
+        addListener(EventPriority.NORMAL, receiveCancelled, consumer);
+    }
+
+    @Override
+    public <T extends Event> void addListener(boolean receiveCancelled, Class<T> eventType, Consumer<T> consumer) {
+        checkNotGeneric(eventType);
+        addListener(EventPriority.NORMAL, receiveCancelled, eventType, consumer);
+    }
+
+    @Override
+    public <T extends Event> void addListener(Class<T> eventType, Consumer<T> consumer) {
+        checkNotGeneric(eventType);
```

Comment:
Please remove the `checkNotGeneric` tests. `addListener(EventPriority, Class, Z, Consumer)` already checks for generic-ity.